File: MIT600/ps1/ProblemSet-1-1.py
balance = float(raw_input('Enter the outstanding balance on the credit card: '))
interestRate = float(raw_input('Enter annual interest rate: '))
minimumRate = float(raw_input('Enter minimum monthly payment rate: '))

remainingBalance = balance

for x in range(12):
    minimumMonthlyPayment = round((minimumRate * remainingBalance), 2)
    interestPaid = round(((interestRate / 12.00) * remainingBalance), 2)
    principalPaid = round((minimumMonthlyPayment - interestPaid), 2)
    remainingBalance = round((remainingBalance - principalPaid), 2)

    print 'MONTH ' + str(x+1) + '\n' + ' Minimum monthly payment: $' + str(minimumMonthlyPayment) + '\n Principle paid: $' + str(principalPaid) + '\n Remaining balance: $' + str(remainingBalance)

File: 600SC/ps1/ProblemSet-1-1.rb
puts "Enter the outstanding balance on the credit card: "
balance = gets.to_f
puts "Enter annual interest rate: "
interestRate = gets.to_f
puts "Enter minimum monthly payment rate: "
minimumRate = gets.to_f

remainingBalance = balance

for x in 0..11 do
  minimumMonthlyPayment = (minimumRate * remainingBalance).round(2)

  interestPaid = ((interestRate / 12.00) * remainingBalance).round(2)

  principalPaid = (minimumMonthlyPayment - interestPaid).round(2)
  remainingBalance = (remainingBalance - principalPaid).round(2)

  puts "MONTH" + (x+1).to_s + "\n" + " Minimum monthly payment: $" + minimumMonthlyPayment.to_s + "\n" + "Principle paid: $" + principalPaid.to_s + "\n" + "Remaining balance: $" + remainingBalance.to_s
end
